src/devices/src/virtio/net/persist.rs

```diff
@@ -23,10 +24,39 @@ use super::{NUM_QUEUES, QUEUE_SIZE};
 use crate::virtio::persist::{Error as VirtioStateError, VirtioDeviceState};
 use crate::virtio::{DeviceState, TYPE_NET};
 
-#[derive(Clone, Versionize)]
+#[derive(Debug, Default, Clone, Versionize)]
 // NOTICE: Any changes to this structure require a snapshot version bump.
 pub struct NetConfigSpaceState {
-    guest_mac: [u8; MAC_ADDR_LEN],
+    #[version(end = 2, default_fn = "def_guest_mac_old")]
+    guest_mac_old: MacAddr,
```

Comment:
I am actually surprised that the deserialization still works after changing the type of `guest_mac_old`. 
I guess this is due to how bincode does the serialization/deserialization. I suppose that `MacAddr` which behind the scenes is a struct with only one field will get deserialized to exactly what that one field contains.
So that 
```
MacAddr {byest [u8; MAC_ADDR_LEN] 
```
is equivalent to `[u8; MAC_ADDR_LEN]`
To be honest, I do not feel comfortable with this change since this implies that we will depend on how bincode works  (which can very well change in future). 
Also, I feel that when doing such changes in structs that get versionized, we should keep the changes as suggestive  as possible (history wise) by keeping old fields exactly as they were before the addition and apply the freedom of naming and type only on new fields.
In  a nutshell, the way I would have done it:

```
NetConfigSpaceState {
guest_mac: [u8; MAC_ADDR_LEN], -> exaclty as before,
guest_mac_opt/guest_mac_new: Option<MacAddr>,
}
```

Reply:
Well, in my opinion, `MacAddr` should not even be a struct (well it will be) but just: `pub struct`

Reply:
Well in my opnion `MacAddr` should be:
```rust
pub struct MacAddr([u8; MAC_ADDR_LEN])
```
instead of what it is not. Then it is clear that it will always be ser/deser as just 6 bytes. And I don't think where is any way this bytes would be ser/de in any other way.

Also about changing the name to `guset_mac` and `guest_mac_opt/_new`, I think this is confusing, because in new versions of FC only `guest_mac_opt/_new` would be used. Then if someone would work with this `NetConfig` it would be unclear what field to use.

Reply:
> Well in my opnion `MacAddr` should be:
> 
> ```rust
> pub struct MacAddr([u8; MAC_ADDR_LEN])
> ```
> 
> instead of what it is now. Then it is clear that it will always be ser/deser as just 6 bytes. And I don't think where is any way this bytes would be ser/de in any other way.
> 
> Also about changing the name to `guset_mac` and `guest_mac_opt/_new`, I think this is confusing, because in new versions of FC only `guest_mac_opt/_new` would be used. Then if someone would work with this `NetConfig` it would be unclear what field to use.

I can only agree that `MacAddr` should have been exactly that, but it s not right now, true? My personal take on this is that I would not choose to change an old field (which indeed was not really the best choice) on which snapshot compatibility depends for the sake of readability? (I guess).

I was saying in my previous comment that we should have the changes to these fields tell a story. Maybe the names suggested by me are not the happiest choice and we can go for some other options (`guest_mac_v2`
etc). Also, If we need to change the type of `guest_mac` in the future, what would we call it that time?

Reply:
Ok, we can have a bit ugly, but very story telling naming like `guest_mac_pre_1_2`, `guest_mac_1_2` and so on.
Also to make old fields more explicit we can use `#[deprecated]`like this:
```rust
 #[deprecated(since = "1.2.0", note = "this field is deprecated")]
 #[version(end = 2, default_fn = "def_guest_mac_old")]
 guest_mac_pre_1_2: [u8; MAC_ADDR_LEN],
 #[version(start = 2, de_fn = "de_guest_mac", ser_fn = "ser_guest_mac")]
 guest_mac_1_2: Option<MacAddr>,
```

Reply:
Indeed it is a bit ugly to add all the number transitions; 
I think that end = 2, start  = 2  suffice to correlate to the version map.
The deprecated thingy I like a lot, since it s a way to also correlate the field to the FC version.

Reply:
Well, lets leave adding deprecation to our future selfs. It deserves a separate PR because there are several places where we updated fields this way.